crates/bevy_ui/src/geometry.rs

```diff
@@ -46,6 +46,8 @@ pub enum Val {
     /// * For `margin`, `padding`, and `border` values: the percentage is relative to the parent node's width.
     /// * For positions, `left` and `right` are relative to the parent's width, while `bottom` and `top` are relative to the parent's height.
     Percent(f32),
+    /// Value relative to the font size of the root element
```

Comment:
Is this truly the root element of the hierarchy, or the default size in the resource? Normally the "r" in "rem" stands for "root" and everyone knows what it means, but perhaps in our case it should stand for "resource".

Reply:
It uses the value of the font size on the root UI node, which is taken from the root nodes TextFont component if it's present, or, if it's not, the DefaultTextStyle resource.

I'm wondering though if "rem" is the right name to use. I think it probably is, but "root element' is more web terminology again, which I want to avoid.  Thought about "rent" for "root entity" but it seems a bit silly.

Reply:
Unfortunately, I think we are dealing with conflicting needs here: a tradeoff between simplicity and flexibility.

Unlike the web where there's a single root element (`html`), in Bevy you have multiple roots. This means that if font size is tied to the UI root, then globally adjusting the font size requires updating every root instead of adjusting a single parameter in a resource.

There's also a concern about cost: for something like an editor or app with nested panels, UI hierarchies can get fairly deep, with a depth of 10-15 levels not being uncommon. Calculating the effective font size for a leaf node could require looking up components on every ancestor.

Reply:
> Maybe we could have a seperate "gem" (or something) unit based on the global font size?

I agree with this design.